active/InformationBar/InformationBar.md

```diff
@@ -0,0 +1,622 @@
+adapt to the various color states and accessibility features.
+
+
+Please view the UX guidance for
+[Standard Icons](https://docs.microsoft.com/en-us/windows/win32/uxguide/vis-std-icons) 
+and [Color](https://docs.microsoft.com/en-us/windows/win32/uxguide/vis-color) 
+to ensure your message is communicated clearly and accessible to users.
+
+
+### Severity
+ Avoid setting the Severity property for a notification that does not match the information communicated in 
+ the Title, Message, or custom content.
+ 
+
+ The accompanying information should aim to communicate the following to use that Severity.
+ - Error: An error or problem that has occurred.
+ - Warning: A condition that might cause a problem in the future.
+ - Success: A long-running and/or background task has completed.
+ - Default: General information that requires the user's attention.
+
+Icons and color should not be the only UI components signifying meaning for your notification. 
+Text in the notification's Title and/or Message should be included to display information.
+
+
+## Message 
+
+Text in your notification will not be a constant length in all languages. For the Title and Message property 
+this may impact whether your notification will expand to a second line. 
+We recommend you avoid positioning based on message length or other UI elements set to a specific language.
+
+The notification will follow standard mirroring behavior when localized to/from languages that are right to left (RTL) or left to right (LTR). The icon will only mirror if there is directionality.
+
+Please view the guidance for 
+[Adjust layout and fonts, and support RTL](https://docs.microsoft.com/en-us/windows/uwp/design/globalizing/adjust-layout-and-fonts--and-support-rtl) 
+for more information about text localization in your notification.
+
+
+# API Notes
+
+### Notable Properties  
+| Name | Description |
+|:-:|:--|
+| IsOpen| Gets or sets a value that determines the visibility of the InfoBar. By default, is set to false. |
+| Severity | Gets or sets a value that indicates the  color and icon to style the InfoBar. It will also define the assistive technology settings. |
+| IsUserDismissable| Gets or sets a boolean that indicates whether the user will be able to dismiss the InfoBar.
+
+
+### Events  
+| Name | Description |
+|:-:|:--|
+| CloseButtonClick | Occurs after the close button has been clicked. |
+| Closed | Occurs after the info bar is closed. |
+| Closing |Occurs just before the info bar begins to close. |
+
+
+# API Details
+
+```c++
+enum InfoBarCloseReason
+{
+    CloseButton = 0,
+    Programmatic = 1,
+};
+
+enum InfoBarSeverity
+{
+    Error = 0,
+    Warning = 1,
+    Success = 2,
+    Default = 3,
+}
+
+runtimeclass InfoBarClosedEventArgs
+{
+    InfoBarCloseReason Reason{ get; };
+};
+
+runtimeclass InfoBarClosingEventArgs
+{
+    InfoBarCloseReason Reason{ get; };
+    Boolean Cancel;
+    Windows.Foundation.Deferral GetDeferral();
```

Comment:
Is there a good scenario for deferral? Is there a deadline for the deferral? Is it any different than the app specifying `Cancel = true`, disabling input on the control, doing the async work, then programmatically closing the bar?

Reply:
One example scenario is if the InfoBar is showing a "Terms and Conditions are updated and need to be accepted" message and there's an inline checkbox that a user needs to check before the InfoBar closes. If the user manually closes the InfoBar but the checkbox isn't checked then the close could be cancelled.

@teaP for the deferral deadline and replicating deferral behavior questions?

Reply:
`Cancel` makes perfect sense; it's the deferral that is potentially not useful. Usually it is used if you need to make a network connection or write to a brokered file.

Reply:
In this case it's to enable a dialog and user interaction before closing the InfoBar, which like files and network is async.